src/main/java/com/mentoring/ecommerce/adapter/out/amazon/AmazonS3Adapter.java

```diff
@@ -0,0 +1,47 @@
+package com.mentoring.ecommerce.adapter.out.amazon;
+
+import com.amazonaws.services.s3.model.Bucket;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.util.IOUtils;
+import com.mentoring.ecommerce.application.port.out.S3HandlerPort;
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.mentoring.configuration.S3Configuration.amazonS3Client;
+
+@Component
+public class AmazonS3Adapter implements S3HandlerPort {
+
+    @Value("${s3.bucket-name}")
+    private String bucketName;
+
+    public List<String> listBuckets() {
```

Comment:
I created a simpler method, `listBucket`, to test the `AWS` connection and left it there.
I'm wondering how the cloud methods should be organized:

- in a single generic class as it is;
- or in a different configuration, each has its class, for example.